spring-cloud-stream-binders/spring-cloud-stream-binder-test/src/main/java/org/springframework/cloud/stream/binder/AbstractBinderTests.java

```diff
@@ -169,7 +170,7 @@ public void testSendAndReceiveNoOriginalContentType() throws Exception {
 		assertNotNull(inbound);
 		assertEquals("foo", inbound.getPayload());
 		assertNull(inbound.getHeaders().get(BinderHeaders.BINDER_ORIGINAL_CONTENT_TYPE));
-		assertNull(inbound.getHeaders().get(MessageHeaders.CONTENT_TYPE));
+		assertEquals(MimeTypeUtils.TEXT_PLAIN_VALUE, inbound.getHeaders().get(MessageHeaders.CONTENT_TYPE));
```

Comment:
This test case exposes a `caveat` where the producer message doesn't actually set the `contentType` which means the `originalContentType` was never set by the producer SCSt app. Even in that case, we would set the `contentType` as the deserialised message `contentType` which I believe doesn't do any harm.